.github/workflows/labeler.yml

```diff
@@ -0,0 +1,46 @@
+---
+# Copyright (C) 2023 Maxwell G <maxwell@gtmx.me>
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+"on":
+  pull_request_target:
+  issues:
+    types:
+      - opened
+
+name: "Triage Issues and PRs"
+
+permissions:
+  issues: write
+  pull-requests: write
+
+jobs:
+  label_prs:
+    runs-on: ubuntu-latest
+    name: "Label Issue/PR"
+    steps:
+      - name: Print event information
+        run: |
+          echo "${{ toJSON(github.event) }}"
+      - name: Checkout parent repository
+        uses: actions/checkout@v3
```

Comment:
How about not checking out the repo from the PR, but the current `devel` branch? That way malicious PRs do not result in malicious code being run.

(It does make it harder to update the script, but it also makes the workflow a lot safer. And this can also run automatically for first-time contributors.)

Reply:
Definitely! The `pull_request_target` trigger already does this. If the PR targets `devel`, `devel` will be checked out. If the PR targets `stable-2.15`, `stable-2.15` will be checked out, and so on.

Reply:
Hmm, but then why do you write

> Note: someone will need to go into https://github.com/ansible/ansible-documentation/settings/actions and make sure `Require approval for first-time contributors who are new to Github` is set.

? Doesn't that *not* automatically run this workflow when new contributors create an issue or PR?

Reply:
That's the most lenient option, unfortunately. The default is to require approval for *any* new contributors.

Reply:
Hmm, in gh.com/ansible-collections this is differently: there by default, `pull_request_target` jobs always run. (That's why docs generation works for new contributors, while other tests don't run.)

Reply:
This seems to be the behavior everywhere according to https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks:

> Note: Workflows triggered by `pull_request_target` events are run in the context of the base branch. Since the base branch is considered trusted, workflows triggered by these events will always run, regardless of approval settings.

Reply:
Cool, thanks for verifying that!